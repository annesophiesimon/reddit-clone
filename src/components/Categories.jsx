const Categories = (subreddit) => {
  const { handleClick, categories } = subreddit;

  return (
    <aside className="shadow-lg rounded-xl md:w-1/6">
      <h1 className="font-bold p-2">Categories</h1>
      <ul className="my-2">
        {categories &&
          categories.map((subreddit) => (
            <li key={subreddit.id} className="p-2 hover:bg-slate-400">
              <button onClick={handleClick} value={subreddit.display_name_prefixed}>
                {subreddit.display_name}
              </button>
            </li>
          ))}
      </ul>
    </aside>
  );
};

export default Categories;
