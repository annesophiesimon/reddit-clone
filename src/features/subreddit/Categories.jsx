import { fetchSubreddit } from './subredditSlice';
import { useAppSelector, useAppDispatch } from '../../app/hooks';
import { useEffect } from 'react';

const Categories = (subreddit) => {
  const { handleClick } = subreddit;
  const { subreddits, isCategoriesLoading } = useAppSelector((state) => state.subreddit);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(fetchSubreddit());
  }, []);

  if (isCategoriesLoading) {
    return <div>Data loading</div>; // Add spinner
  }
  return (
    <aside className="shadow-lg border border-grey-50 rounded-xl min-w-[300px] h-full">
      <h1 className="font-bold p-5">Categories</h1>
      <ul className="my-2">
        {subreddits &&
          subreddits
            .filter((subreddit) => subreddit.display_name !== 'AskReddit')
            .map((subreddit, index) => (
              <li key={index} className="px-5 py-2 hover:bg-slate-500/[.06]">
                <button onClick={handleClick} value={subreddit.display_name_prefixed}>
                  {subreddit.display_name}
                </button>
              </li>
            ))}
      </ul>
    </aside>
  );
};

export default Categories;
