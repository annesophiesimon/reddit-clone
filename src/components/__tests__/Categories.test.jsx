import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Categories from '../Categories';

describe('Categories', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Categories />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Post properly', () => {
    render(<Categories title="title" />);
    const titleElement = screen.getByRole('heading', {
      name: /Categories/i
    });
    expect(titleElement).toBeInTheDocument();
  });
});
