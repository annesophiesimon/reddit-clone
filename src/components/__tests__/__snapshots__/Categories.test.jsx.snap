// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<aside
  className="shadow-lg rounded-xl md:w-1/6"
>
  <h1
    className="font-bold p-2"
  >
    Categories
  </h1>
  <ul
    className="my-2"
  />
</aside>
`;
